src/sync/update_tx.js

```diff
@@ -1,157 +1,214 @@
+          resultIndex: tx.optionIdx,
+          senderAddress: tx.senderAddress,
+        });
+        txid = setResultTx.txid;
+      } catch (err) {
+        logger.error(`Error calling CentralizedOracle.setResult: ${err.message}`);
+        throw err;
+      }
 
-    try {
-      await Transactions.insert(tx);
-    } catch (err) {
-      logger.error(`Error insert Transactions: ${err.message}`);
-      throw err;
+      await DBHelper.insertTransaction(Transactions, {
+        _id: txid,
+        txid,
+        version: tx.version,
+        type: 'SETRESULT',
+        status: 'PENDING',
+        senderAddress: tx.senderAddress,
+        oracleAddress: tx.oracleAddress,
+        optionIdx: tx.optionIdx,
+        token: 'BOT',
+        amount: tx.amount,
+        createdTime: Date.now().toString(),
+      });
+      break;
     }
-  } else if (approveTx.type === 'APPROVEVOTE') {
-    let voteTxid;
-    try {
-      const resp = await fetch('http://localhost:5555/vote', {
-        method: 'POST',
-        body: JSON.stringify({
-          contractAddress: approveTx.entityId,
-          resultIndex: approveTx.optionIdx,
-          botAmount: approveTx.amount,
-          senderAddress: approveTx.senderAddress,
-        }),
-        headers: { 'Content-Type': 'application/json' },
-      }).then(res => res.json());
-      voteTxid = resp.result.txid;
-    } catch (err) {
-      logger.error(`Error call /vote: ${err.message}`);
-      throw err;
+
+    // Approve was reset to 0. Sending approve for vote amount.
+    case 'RESETAPPROVEVOTE': {
+      try {
+        const approveTx = await bodhiToken.approve({
+          spender: tx.topicAddress,
+          value: tx.amount,
+          senderAddress: tx.senderAddress,
+        });
+        txid = approveTx.txid;
+      } catch (err) {
+        logger.error(`Error calling BodhiToken.approve: ${err.message}`);
+        throw err;
+      }
+
+      await DBHelper.insertTransaction(Transactions, {
+        _id: txid,
+        txid,
+        version: tx.version,
+        type: 'APPROVEVOTE',
+        status: 'PENDING',
+        senderAddress: tx.senderAddress,
+        topicAddress: tx.topicAddress,
+        oracleAddress: tx.oracleAddress,
+        optionIdx: tx.optionIdx,
+        token: 'BOT',
+        amount,
+        createdTime: Date.now().toString(),
+      });
+      break;
     }
 
-    const tx = {
-      _id: voteTxid,
-      version: approveTx.version,
-      type: 'VOTE',
-      txStatus: 'PENDING',
-      senderAddress: approveTx.senderAddress,
-      entityId: approveTx.entityId,
-      optionIdx: approveTx.optionIdx,
-      token: 'BOT',
-      amount: approveTx.approveTx.amount,
-      createdTime: Date.now().toString(),
-    };
+    // Approve was accepted. Sending vote.
+    case 'APPROVEVOTE': {
+      try {
+        const voteTx = await decentralizedOracle.vote({
+          contractAddress: tx.oracleAddress,
+          resultIndex: tx.optionIdx,
+          botAmount: tx.amount,
```

Comment:
add `0x` as tx.amount will be hexstring

Reply:
this is actually the tx that we construct (as opposed to the tx from searchlogs), so it is already a decimal string.